src/Runtime/LambdaRuntime.php

```diff
@@ -42,19 +43,33 @@ final class LambdaRuntime
     /** @var Invoker */
     private $invoker;
 
-    public static function fromEnvironmentVariable(): self
+    /** @var int seconds */
+    private $timeout;
+
+    public static function fromEnvironmentVariable(?int $timeout = null): self
```

Comment:
I'm not sure we need to make the timeout configurable. After all, since it's based on the Lambda timeout users can customize the timeout via Lambda's configuration.

How about a boolean: enabled (true/false)?

Reply:
I did a boolean at first. But that did not allow me to always use a Lambda timeout of 28 seconds and just "play" with `BREF_TIMEOUT`. 

The edge case is that my "shutdown process" may take 5 seconds.

Reply:
Not sure I see what you mean by "shutdown process"?

> But that did not allow me to always use a Lambda timeout of 28 seconds and just "play" with BREF_TIMEOUT.

I don't think that's something we need to support. If I understand correctly, that would mean the timeout is entirely controlled by the Lambda timeout, which sounds fine to me. Less settings, less code paths, less options is better. At least unless we have a clear use case for allowing 2 different controls (Lambda timeout + env variable)?

Reply:
Forgot about the fact that at first it's good to make the feature opt-in (i.e. it's a bit "experimental" until we enable it by default).

So how about:

- `BREF_TIMEOUT` undefined -> no Bref timeout
- `BREF_TIMEOUT=1` -> automatic Bref timeout

If all works as expected, we can them remove that feature flag and enable it by default in the future?

I just don't see a use case for a custom timeout on top of the already customisable Lambda timeout.

Reply:
> Not sure I see what you mean by "shutdown process"?

Whenever an exception is thrown. My application will flush the logs, maybe write something to the DB and maybe send an email to admin. This could take 3 seconds. This is my "shutdown process". If we don't allow to customise the time to when we thrown an exception, then I cannot use this feature. 

-------

> Less settings, less code paths, less options is better.

We still need a way to opt-out from this feature, right? If so, we still need an environment variable. 

------

Sure let's do it as a feature flag to opt-in to this feature in version 1.3. But I would do 

* `BREF_TIMEOUT` undefined => no Bref timeout
* `BREF_TIMEOUT=0` => automatic Bref timeout
* `BREF_TIMEOUT=-1` => no Bref timeout

Then in version 1.x
* `BREF_TIMEOUT` undefined => automatic Bref timeout